kubectl-plugin/pkg/cmd/create/create_cluster.go

```diff
@@ -0,0 +1,223 @@
+package create
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/generation"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type CreateClusterOptions struct {
+	configFlags    *genericclioptions.ConfigFlags
+	ioStreams      *genericclioptions.IOStreams
+	clusterName    string
+	fileName       string
+	rayVersion     string
+	image          string
+	headCPU        string
+	headMemory     string
+	workerGrpName  string
+	workerCPU      string
+	workerMemory   string
+	workerReplicas int32
+}
+
+var (
+	createClusterLong = templates.LongDesc(`
+		Creates Ray Cluster from inputed file or generate one for user.
+	`)
+
+	createClusterExample = templates.Examples(`
+		# Creates Ray Cluster from inputted Yaml
+		kubectl ray create cluster -f raycluster-stample.yaml
+
+		# Creates Ray Cluster from flags input
+		kubectl ray create cluster sample-cluster --ray-version 2.9.0 --image rayproject/ray:2.9.0 --head-cpu 1 head-memory 5Gi --worker-grp-name worker-group1 --worker-replicas 3 --worker-cpu 1 --worker-memory 5Gi
+	`)
+)
+
+func NewCreateClusterOptions(streams genericclioptions.IOStreams) *CreateClusterOptions {
+	return &CreateClusterOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+		ioStreams:   &streams,
+	}
+}
+
+func NewCreateClusterCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewCreateClusterOptions(streams)
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:          "cluster [CLUSTERNAME] [-f/--filename FILE_PATH]",
+		Short:        "Create Ray Cluster resource",
+		Long:         createClusterLong,
+		Example:      createClusterExample,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := options.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			return options.Run(cmd.Context(), cmdFactory)
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.fileName, "filename", "f", options.fileName, "Path and name of the Ray Cluster yaml file")
+	cmd.Flags().StringVar(&options.rayVersion, "ray-version", options.rayVersion, "Ray Version to use in the Ray Cluster yaml")
+	cmd.Flags().StringVar(&options.image, "image", options.image, "Ray image to use in the Ray Cluster yaml")
+	cmd.Flags().StringVar(&options.headCPU, "head-cpu", options.headCPU, "Number of CPU for the ray head")
```

Comment:
consider some reasonable defaults for head-cpu, head-memory, worker-cpu and worker-memory

Reply:
Default is 500m for cpu and 1Gi for memory.